src/macros.rs

```diff
@@ -0,0 +1,9 @@
+#[macro_use]
+
+macro_rules! map {
+    ( $( $k:expr => $v:expr ),+ ) => {{
+        let mut map = ::std::collections::HashMap::new();
+        $( map.insert($k, $v.to_owned()); )+
```

Comment:
Can I get an explanation of what this map is and what we use it for?

Reply:
This is just a macro that allows us to declare `HashMap` literals, i.e.
```rs
let map = map!{ 0 => 'a', 1 => 'b', 4 => 'd' };
```
rather than
```rs
let mut map = HashMap::new();
map.insert(0, 'a');
map.insert(1, 'b');
map.insert(4, 'd');
```

Reply:
Seems handy.